packages/versioning/src/versioning.ts

```diff
@@ -447,8 +488,7 @@ export function removedOnOrBefore(p: Program, type: Type, version: ObjectType) {
 }
 
 export function renamedAfter(p: Program, type: Type, version: ObjectType) {
```

Comment:
I would change the name, as this doesn't make much sense anymore.

```suggestion
export function hasDifferentNameAtVersion(p: Program, type: Type, version: ObjectType) {
```

Reply:
I wouldn't care here, it is a bug to me as its broken right now. Also projections are explicitly experimental for now so wouldn't worry too much

Reply:
if we reallt don't want breaking here I would keep the older method as it was(bugged) and marked it deprecated and have the new one be the one with the fix

Reply:
I preserved the old method and marked it deprecated, but none of the versioning cadl references the old name, so it only exists for emitters that might have been using it.